backend/src/packages/chaiNNer_standard/utility/directories/into_directory.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+import sys
+from pathlib import Path
+
+from nodes.properties.inputs import DirectoryInput, TextInput
+from nodes.properties.outputs import DirectoryOutput
+
+from .. import value_group
+
+separator = "/" if sys.platform == "linux" else r"\\"
+
+
+@value_group.register(
+    schema_id="chainner:utility:into_directory",
+    name="Into Directory",
+    description="Goes forward into a directory.",
+    icon="BsFolder",
+    inputs=[
+        DirectoryInput(
+            "Directory", must_exist=False, label_style="hidden", has_handle=True
+        ),
+        TextInput("Folder", has_handle=True),
+    ],
+    outputs=[
+        DirectoryOutput(
+            "Directory",
+            output_type='Directory { path: string::concat(Input0.path, "'
+            + separator
+            + '", Input1) }',
+        ),
+    ],
+)
+def into_directory_node(directory: Path, folder: str) -> Path:
+    return directory / folder
```

Comment:
This has issues with path traversal. Example:
```py
>>> Path("C:/foo/bar") / "/bar"
WindowsPath('C:/bar')
>>> Path("C:/foo/bar") / "D:/bar"
WindowsPath('D:/bar')
>>> Path("C:/foo/bar") / "../"
WindowsPath('C:/foo/bar/..') 
# ".." is not a valid dir name, so some libraries may resolve the path before. 
# This could lead to parts of chainner interpreting paths differently, which can lead to potential security vulnerabilities.
>>> (Path("C:/foo/bar") / "../").parent
WindowsPath('C:/foo/bar')
```

We need to define proper behavior for all these cases. The first question is whether we even want to allow `..` and absolute paths.

Reply:
I believe with my most recent push all of these are now consistent between the python value and navi type and work as you would expect.

Reply:
... You solve one thing. Since you `.resolve()` now, it does the path traversal for `..`.

But my question was whether we should even allow that. Same for absolute paths.

Reply:
... You solve one thing. Since you `.resolve()` now, it does the path traversal for `..`.

But my question was whether we should even allow that. Same for absolute paths.

Reply:
I think so, because it mirrors the behavior of both javascript and python for how they combine paths. Why would we make this different if there's a "standard" for resolving a valid path out of combining path strings like this?

Reply:
Because they are joining paths. They don't call these functions "go into directory".

Reply:
Sure yeah, but we also don't have to call it that. I would personally think it would be far more convenient for someone who does know how paths work to be able to pass in a directory, type `../../../whatever/whatever` and have it work, while also retaining the ability to go up _x_ directories in a dedicated node.

Reply:
Sure. Then let's support `..`. But I don't see how joining with absolute paths would be useful.

Reply:
> But I don't see how joining with absolute paths would be useful.

These are not absolute paths

Reply:
See the following examples:

```py
>>> Path("C:/foo/bar") / "/bar"
WindowsPath('C:/bar')
>>> Path("C:/foo/bar") / "D:/bar"
WindowsPath('D:/bar')
```

The string value this node gets can very well represent an absolute path.

Reply:
Ok, then what do you suggest we do? restrict it to just single folder names?

Reply:
> restrict it to just single folder names?

We can't do that since nodes like Load Images can return multiple levels as their subdirectory path.

I guess we should just disallow absolute paths as the join argument. Unfortunately, python doesn't consider `/bar` an absolute path on windows. This is consistent with how windows itself treats paths, but it doesn't make sense for a cross-platform API. So we have to handle this ourselves.